src/main/java/org/folio/circulation/domain/OpeningPeriod.java

```diff
@@ -0,0 +1,32 @@
+package org.folio.circulation.domain;
+
+import io.vertx.core.json.JsonObject;
+import org.joda.time.DateTime;
+import org.joda.time.LocalDate;
+
+public class OpeningPeriod {
+  private static final String DATE_KEY = "date";
+
+  private final OpeningDay openingDay;
+  private final LocalDate date;
+
+  public OpeningDay getOpeningDay() {
+    return openingDay;
+  }
+
+  public LocalDate getDate() {
+    return date;
+  }
+
+  public OpeningPeriod(LocalDate date, OpeningDay openingDay) {
+    this.openingDay = openingDay;
+    this.date = date;
+  }
+
+  public OpeningPeriod(JsonObject jsonObject) {
+    this.openingDay = OpeningDay.fromJsonByDefaultKey(jsonObject);
+    String dateProperty = jsonObject.getString(DATE_KEY);
+    this.date = DateTime.parse(dateProperty).toLocalDate();
+  }
```

Comment:
This code does not appear to be covered by any tests any more, does that mean that it can be removed (similarly for the `getDate` method above)?

Reply:
It is still used during deserialization in `CalendarRepository`.

Reply:
Is the code within the calendar repository used? SonarQube was reporting that this code was not covered by any of the tests.

Reply:
It is used, but it's not covered with tests. As we agreed earlier, it will be covered with API tests once this feature is integrated.

Reply:
Ok, personally, I would remove the code until it is exercised in the system. I'll resolve this, and leave the decision to you.